run.c

```diff
@@ -339,12 +339,29 @@ human_interaction(double TimeLastPM, double TimeLastOut)
  * be outside the allowed bounds, it will be readjusted by the function ensure_neighbours(), and for those
  * particle, the densities are recomputed accordingly. Finally, the hydrodynamical forces are added.
  */
-void compute_accelerations(int is_PM, int FirstStep)
+void compute_accelerations(int is_PM, int FirstStep, int GasEnabled)
 {
-    message(0, "Start force computation...\n");
+    message(0, "Begin force computation.\n");
 
     walltime_measure("/Misc");
 
+    /* We do this first so that the density is up to date for
+     * adaptive gravitational softenings. */
+    if(GasEnabled > 0)
+    {
+        /***** density *****/
+        message(0, "Start density computation...\n");
+
+        density();		/* computes density, and pressure */
+
+        /***** update smoothing lengths in tree *****/
+        force_update_hmax(ActiveParticle, NumActiveParticle);
+        /***** hydro forces *****/
+        message(0, "Start hydro-force computation...\n");
+
+        hydro_force();		/* adds hydrodynamical accelerations  and computes du/dt  */
```

Comment:
Looks like hydroforce can go after gravity? Gravity doesn't depend on it at all.

Reply:
It could - I just thought conceptually density + hydro go together and the sfr/bh go together too. But not very important.